clients/common/norm.cpp

```diff
@@ -1,596 +0,0 @@
-/* ************************************************************************
```

Comment:
I agree with moving `norm.cpp` to a header file `norm.hpp`.

Reply:
Thanks for agreement  :)